e2e/tcproute_test.go

```diff
@@ -441,3 +443,231 @@ func defaultNetworkURL() string {
+	if err != nil {
+		t.Fatalf(fmt.Sprintf("buildTcpRoute(_, %s, multiple-tcpr-0, %v, %s) = %v, want nil", meshURL, rules, bss[1], err))
+	}
+	toRemove := graph.Get(bss[0])
+	toRemove.Builder().SetState(rnode.NodeDoesNotExist)
+	removedBS, bss := bss[0], bss[1:]
+	graph, err = graphBuilder.Build()
+	if err != nil {
+		t.Fatalf("graphBuilder.Build() = %v, want nil", err)
+	}
+	result, err = plan.Do(ctx, theCloud, graph)
+	if err != nil {
+		t.Fatalf("plan.Do(_, _, _) = %v, want nil", err)
+	}
+
+	expectedActions := []exec.ActionMetadata{
+		{Type: exec.ActionTypeMeta, Name: eventName(removedBS)},
+		{Type: exec.ActionTypeMeta, Name: eventName(bss[0])},
+		{Type: exec.ActionTypeMeta, Name: eventName(bss[1])},
+		{Type: exec.ActionTypeMeta, Name: eventName(bss[2])},
+		{Type: exec.ActionTypeMeta, Name: eventName(bss[3])},
+		{Type: exec.ActionTypeMeta, Name: eventName(bss[4])},
+		{Type: exec.ActionTypeMeta, Name: eventName(hcs[0])},
+		{Type: exec.ActionTypeMeta, Name: eventName(hcs[1])},
+		{Type: exec.ActionTypeMeta, Name: eventName(hcs[2])},
+		{Type: exec.ActionTypeMeta, Name: eventName(hcs[3])},
+		{Type: exec.ActionTypeMeta, Name: eventName(hcs[4])},
+		{Type: exec.ActionTypeMeta, Name: eventName(hcs[5])},
+		{Type: exec.ActionTypeUpdate, Name: actionName(exec.ActionTypeUpdate, tcprs[0])},
+		{Type: exec.ActionTypeUpdate, Name: actionName(exec.ActionTypeUpdate, tcprs[1])},
+		{Type: exec.ActionTypeUpdate, Name: actionName(exec.ActionTypeUpdate, tcprs[2])},
+	}
+
+	processGraphAndExpectActions(t, graphBuilder, expectedActions)
+
+	toRemove = graph.Get(tcprs[0])
+	toRemove.Builder().SetState(rnode.NodeDoesNotExist)
+	removedTcpr, tcprs := tcprs[0], tcprs[1:]
+	graph, err = graphBuilder.Build()
+	if err != nil {
+		t.Fatalf("graphBuilder.Build() = %v, want nil", err)
+	}
+	result, err = plan.Do(ctx, theCloud, graph)
+	if err != nil {
+		t.Fatalf("plan.Do(_, _, _) = %v, want nil", err)
+	}
+	expectedActions = []exec.ActionMetadata{
+		{Type: exec.ActionTypeMeta, Name: eventName(removedBS)},
+		{Type: exec.ActionTypeMeta, Name: eventName(bss[0])},
+		{Type: exec.ActionTypeMeta, Name: eventName(bss[1])},
+		{Type: exec.ActionTypeMeta, Name: eventName(bss[2])},
+		{Type: exec.ActionTypeMeta, Name: eventName(bss[3])},
+		{Type: exec.ActionTypeMeta, Name: eventName(bss[4])},
+		{Type: exec.ActionTypeMeta, Name: eventName(hcs[0])},
+		{Type: exec.ActionTypeMeta, Name: eventName(hcs[1])},
+		{Type: exec.ActionTypeMeta, Name: eventName(hcs[2])},
+		{Type: exec.ActionTypeMeta, Name: eventName(hcs[3])},
+		{Type: exec.ActionTypeMeta, Name: eventName(hcs[4])},
+		{Type: exec.ActionTypeMeta, Name: eventName(hcs[5])},
+		{Type: exec.ActionTypeUpdate, Name: actionName(exec.ActionTypeUpdate, removedTcpr)},
+		{Type: exec.ActionTypeUpdate, Name: actionName(exec.ActionTypeUpdate, tcprs[0])},
+		{Type: exec.ActionTypeUpdate, Name: actionName(exec.ActionTypeUpdate, tcprs[1])},
+	}
+
+	processGraphAndExpectActions(t, graphBuilder, expectedActions)
```

Comment:
you should check in GCE that tcpRoute with it's backends  was deleted

Reply:
Done.